File: Programmers/python/level1/K번째수.py
# https://programmers.co.kr/learn/courses/30/lessons/42748

def solution(array, commands):
    answer = []
    for i in commands:
        arr = array[i[0]-1:i[1]]
        arr.sort()
        answer.append(arr[i[2]-1])
    return answer

a = [1, 5, 2, 6, 3, 7, 4]
b = [[2, 5, 3], [4, 4, 1], [1, 7, 3]]
print(solution(a, b))

# 더 간결한 코드

# def solution(array, commands):
#     return list(map(lambda x:sorted(array[x[0]-1:x[1]])[x[2]-1], commands))

# lambda의 사용방법 : https://dojang.io/mod/page/view.php?id=2359

File: Programmers/Java_lan/level1/K번째수.java
import java.util.*;

public class K번째수 {
    public static void main(String[] args) {
        int[] array = {1, 5, 2, 6, 3, 7, 4};
        int[][] commands = {{2, 5, 3}, {4, 4, 1}, {1, 7, 3}};
        int[] answer = solution(array, commands);
        for(int a : answer) {
            System.out.print(a + " ");
        }
    }

    public static int[] solution(int[] array, int[][] commands) {        
        int[] answer = new int[commands.length];
        for(int i = 0; i < commands.length; i++) {
            int[] ary = Arrays.copyOfRange(array, commands[i][0] - 1, commands[i][1]);
            Arrays.sort(ary);
            answer[i] = ary[commands[i][2] - 1];
        }
        return answer;
    }
}
